doc/book/helpers/server-url-helper.md

```diff
@@ -78,20 +79,23 @@ $container->set(
 );
 ```
 
-To register the `ServerUrlMiddleware` as pre-routing pipeline middleware:
+To register the `ServerUrlMiddleware` as pipeline middleware following the
+routing middleware:
 
 ```php
 use Zend\Expressive\Helper\ServerUrlMiddleware;
 
-// Do this early, before piping other middleware or routes:
+// Programmatically:
+$app->pipeRoutingMiddleware();
 $app->pipe(ServerUrlMiddleware::class);
```

Comment:
I think it should be piped **before** routing, so that if some error occurs during the routing phase, the error middlewares can still take advantage of it, for example a 404 error template.